qbit-http-storages/src/commonMain/kotlin/qbit.storage/YandexDiskStorage.kt

```diff
@@ -0,0 +1,250 @@
+            client.put<String>(YANDEX_DISK_API_RESOURCES) {
+                parameter("path", path)
+                header("Authorization", "OAuth $accessToken")
+            }
+        }
+    }
+
+    private suspend fun getFile(client: HttpClient, downloadUrl: String): ByteArray {
+        val httpResponse = client.get<HttpResponse>(downloadUrl)
+        return httpResponse.readBytes()
+    }
+
+    private suspend fun getUrlToDownload(client: HttpClient, path: String): HttpResponse {
+        return client.get(YANDEX_DISK_API_GET_DOWNLOAD_FILE_URL) {
+            parameter("path", path)
+            header("Authorization", "OAuth $accessToken")
+        };
+    }
+
+    /**    Helper method for wrapping dirNames to Namespace Collection */
+    private fun wrapDirNamesToNamespaces(dirNames: List<String>, namespace: Namespace): Collection<Namespace> {
+        return dirNames.map { dirName -> Namespace(namespace, dirName) }
+    }
+
+    /**    Helper method for wrapping fileNames to Keys Collection */
+    private fun wrapFileNamesToKeys(fileNames: List<String>, namespace: Namespace): Collection<Key> {
+        return fileNames.map { filename -> Key(namespace, filename) }
+    }
+
+    /**    Helper method for getting file names by type */
+    private fun getFileNamesInResourceByType(resource: Resource, type: String): List<String> {
+        val items = resource._embedded?.items
+        var files: List<String> = emptyList()
+        if (items != null) {
+            files = items.filter { item -> item.type == type }.map { item -> item.name }
+        }
+        return files
+    }
+
+    /**    Helper method for getting full path from namespace to root */
+    private fun getFullPath(namespace: Namespace): String {
+        var parts = namespace.parts
+        return parts.joinToString("/") + "/"
+    }
+
+    /**    Helper method for getting paths from namespace and its parents */
+    private fun generatePutResourcePaths(namespace: Namespace): List<String> {
+        val partsArrayList = arrayListOf<List<String>>()
+        val pathsArrayList = arrayListOf<String>()
+        var currentNamespace = namespace
+        while (currentNamespace.parent != null) {
+            partsArrayList.add(currentNamespace.parts)
+            currentNamespace = currentNamespace.parent!!
+        }
+
+        val iterator = partsArrayList.iterator();
+        while (iterator.hasNext()) {
+            val parts = iterator.next()
+            pathsArrayList.add(parts.joinToString("/") + "/")
+        }
+        return pathsArrayList.reversed()
+    }
+
+    /**    Helper method for getting full path to key, including key name */
+    private fun getFullPath(key: Key): String {
+        val fileName = key.name
+        val namespace = key.ns
+        val parts = namespace.parts
+        val filePath = parts.joinToString("/")
+        return "$filePath/$fileName"
+    }
+}
+
+@InternalCoroutinesApi
+fun <T> runBlocking(body: suspend () -> T): T {
```

Comment:
Create platform qbit-core/src/commonMain/kotlin/qbit/platform/Coroutines.kt and move this method there as expected method, then implement it the same way for jvm

Reply:
expect function

Reply:
expect common
impl jvm
multiplatform projects